backend/src/nodes/nodes/image_filter/dither_palette.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+from enum import Enum
+
+import numpy as np
+
+from . import category as ImageFilterCategory
+from ...impl.dithering.color_distance import batch_nearest_palette_color
+from ...impl.dithering.constants import ErrorDiffusionMap, ERROR_PROPAGATION_MAP_LABELS
+from ...impl.dithering.diffusion import palette_error_diffusion_dither
+from ...impl.dithering.riemersma import palette_riemersma_dither
+from ...node_base import NodeBase, group
+from ...node_factory import NodeFactory
+from ...properties import expression
+from ...properties.inputs import ImageInput, EnumInput, NumberInput
+from ...properties.outputs import ImageOutput
+
+
+class PaletteDitherAlgorithm(Enum):
+    NONE = "None"
+    DIFFUSION = "Diffusion"
+    RIEMERSMA = "Riemersma"
+
+
+PALETTE_DITHER_ALGORITHM_LABELS = {
+    PaletteDitherAlgorithm.NONE: "No dithering",
+    PaletteDitherAlgorithm.DIFFUSION: "Error Diffusion",
+    PaletteDitherAlgorithm.RIEMERSMA: "Riemersma Dithering",
+}
+
+
+@NodeFactory.register("chainner:image:palette_dither")
+class PaletteDitherNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Apply one of a variety of dithering algorithms using colors from a given palette. (Only the top row of pixels (y=0) of the palette will be used.)"
+        self.inputs = [
+            ImageInput(),
+            ImageInput(label="LUT", image_type=expression.Image(channels_as="Input0")),
```

Comment:
This should probably be called "Palette" instead.

Reply:
Okay.  I was trying to have some consistency with the "Apply LUT" node, but I guess a LUT and a Palette are actually different things, even if they're both 1 row images.

Reply:
They are both look up tables, but I think that have slightly different purposes. The LUTs for Apply LUT are essentially gradient maps, so the order of colors matters. But with palettes, the order of colors doesn't matter, right?

Reply:
Yeah, that's a good distinction.